src/test/scala/io/iohk/ethereum/jsonrpc/EthProofServiceSpec.scala

```diff
@@ -0,0 +1,158 @@
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+class EthProofServiceSpec
+    extends TestKit(ActorSystem("EthGetProofSpec_ActorSystem"))
+    with AnyFlatSpecLike
+    with WithActorSystemShutDown
+    with Matchers
+    with ScalaFutures
+    with OptionValues
+    with MockFactory
+    with NormalPatience
+    with TypeCheckedTripleEquals {
+
+  "EthProofService" should "handle getStorageAt request" in new TestSetup {
+    // given
+    val address = Address(ByteString(Hex.decode("abbb6bebfa05aa13e908eaa492bd7a8343760477")))
+
+    val key = 333
+    val value = 123
+
+    val storageMpt = EthereumUInt256Mpt
+      .storageMpt(
+        ByteString(MerklePatriciaTrie.EmptyRootHash),
+        storagesInstance.storages.stateStorage.getBackingStorage(0)
+      )
+      .put(UInt256(key), UInt256(value))
+
+    val account = Account(
+      nonce = 0,
+      balance = UInt256(0),
+      storageRoot = ByteString(storageMpt.getRootHash),
+      codeHash = ByteString("")
+    )
+
+    import MerklePatriciaTrie.defaultByteArraySerializable
+    val mpt =
+      MerklePatriciaTrie[Array[Byte], Account](storagesInstance.storages.stateStorage.getBackingStorage(0))
+        .put(
+          crypto.kec256(address.bytes.toArray[Byte]),
+          account
+        )
+
+    val blockToRequest = Block(Fixtures.Blocks.Block3125369.header, Fixtures.Blocks.Block3125369.body)
+    val newBlockHeader = blockToRequest.header.copy(stateRoot = ByteString(mpt.getRootHash))
+    val newblock = blockToRequest.copy(header = newBlockHeader)
+    blockchain.storeBlock(newblock).commit()
+    blockchain.saveBestKnownBlocks(newblock.header.number)
+
+    val ethGetProof = new EthProofService(blockchain, blockGenerator, blockchainConfig.ethCompatibleStorage)
+    val storageKeys = Seq(StorageProofKey(key))
+    val blockNumber = BlockParam.Latest
+
+    // when
+    val result = ethGetProof.run(address, storageKeys, blockNumber)
+
+    // then
+    val balanceResponse: GetBalanceResponse = ethService
+      .getBalance(GetBalanceRequest(address, BlockParam.Latest))
+      .runSyncUnsafe()
+      .getOrElse(fail("ethService.getBalance did not get valid response"))
+
+    val transactionCountResponse = ethService
+      .getTransactionCount(GetTransactionCountRequest(address, BlockParam.Latest))
+      .runSyncUnsafe()
+      .getOrElse(fail("ethService.getTransactionCount did not get valid response"))
+
+    val storageValues: Seq[ByteString] = storageKeys.map { position =>
+      ethService
+        .getStorageAt(GetStorageAtRequest(address, position.v, BlockParam.Latest))
+        .runSyncUnsafe()
+        .getOrElse(fail("ethService.getStorageAt did not get valid response"))
+        .value
+    }
+
+    val givenResult = result.runSyncUnsafe().getOrElse(fail())
+
+    givenResult.address shouldBe address
```

Comment:
would it be possible to use diffx `matchTo`? it's already in the project and gives a better output.

Reply:
I was not aware of this. Thanks :)
Done.